app/payment-link-v2/reference-confirm/reference-confirm.controller.js

```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+const { response } = require('../../utils/response')
+const { getSessionVariable } = require('../../utils/cookie')
+const { NotFoundError } = require('../../errors')
+const { paymentLinksV2 } = require('../../paths')
+const replaceParamsInPath = require('../../utils/replace-params-in-path')
+
+function getPage (req, res, next) {
+  const product = req.product
+
+  const data = {
+    productExternalId: product.externalId,
+    productName: product.name,
+    paymentReferenceLabel: product.reference_label,
+    paymentReferenceHint: product.reference_hint
+  }
+
+  if (!product.reference_enabled) {
+    return next(new NotFoundError('Attempted to access reference confirm page with a product that auto-generates references.'))
+  }
+
+  const sessionReferenceNumber = getSessionVariable(req, 'referenceNumber')
+
+  data.backLinkHref = replaceParamsInPath(paymentLinksV2.referenceConfirm, product.externalId)
```

Comment:
It's unclear from the designs whether there should be a back link. It probably does make sense to have one, for consistency. But the target URL should be the URL for the reference page.

Reply:
Good spot - thanks